src/librepgp/stream-armor.cpp

```diff
@@ -193,10 +194,12 @@ armor_read_trailer(pgp_source_t *src)
     }
 
     stlen = strlen(param->armorhdr);
-    strncpy(st, ST_ARMOR_END, 8); /* 8 here is mandatory */
-    strncpy(st + 8, param->armorhdr + 5, stlen - 5);
-    strncpy(st + stlen + 3, ST_DASHES, 5);
-    stlen += 8;
+    if (stlen+8+1 <= sizeof(st)) {
```

Comment:
1. Missing spaces around '+' (please do not forget to run clang-format so code looks consistent)
2. Actually if stlen + 8 + 1 is larger than sizeof(st) then there is some problem with code so error message should be logged.